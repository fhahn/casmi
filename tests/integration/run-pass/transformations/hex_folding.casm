// transform, cmdline "--fcf"

CASM hf

init foo

rule foo = 
{
    print hex( 1234 )   // 0x4d2

    program(self) := undef
}
