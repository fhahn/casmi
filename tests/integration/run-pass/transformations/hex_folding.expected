foo=STATEMENTS(LET CONST(ident c0__,str 4d2),STATEMENTS(LET CONST(ident c1__,SELF),STATEMENTS(LET CONST(ident c2__,UNDEF),PARBLOCK(STATEMENTS(PRINT(LIST(ident c0__,null),null),STATEMENTS(UPDATE(LOOKUP(ident program,LIST(ident c1__,null)),ident c2__),null)),null))))
